pkg/storage/action_cache_asset_store.go

```diff
@@ -174,10 +189,24 @@ func (rs *actionCacheAssetStore) Put(ctx context.Context, ref *asset.AssetRefere
 	if err != nil {
 		return err
 	}
-	// Create the action with the qualifier directory as the input root
-	action, command, err := assetReferenceToAction(ref, directoryDigest)
-	if err != nil {
-		return err
+	var action *remoteexecution.Action
+	var command *remoteexecution.Command
+	if commandGenerator, err := qualifier.QualifiersToCommand(ref.Qualifiers); err != nil || len(ref.Uris) > 1 {
+		// Create the action with the qualifier directory as the input root
+		action, command, err = assetReferenceToAction(ref, directoryDigest)
+		if err != nil {
+			return err
+		}
+	} else {
+		command = commandGenerator(ref.Uris[0])
+		commandDigest, err := ProtoToDigest(command)
+		if err != nil {
+			return err
+		}
+		action = &remoteexecution.Action{
+			CommandDigest:   commandDigest,
+			InputRootDigest: directoryDigest,
```

Comment:
As mentioned in an above comment, this should be `EmptyDigest` in place of `directoryDigest`.